shoes-swt/spec/shoes/swt/arc_painter_spec.rb

```diff
@@ -9,72 +9,207 @@
-
-    let(:shape) { Shoes::Swt::Arc.new(dsl, swt_app) }
-    subject { Shoes::Swt::ArcPainter.new(shape) }
-
-    it_behaves_like "stroke painter"
-    it_behaves_like "fill painter"
-
-    context "normal fill style" do
-      before :each do
-        allow(shape).to receive_messages(wedge?: false)
-      end
-
-      specify "fills arc using path" do
-        expect(gc).to receive(:fill_path)
-        subject.paint_control(event)
-      end
-
-      specify "draws arc" do
-        expect(gc).to receive(:draw_arc)
-        subject.paint_control(event)
-      end
-
-      # Swt measures the arc counterclockwise, while Shoes measures it clockwise.
-      specify "translates DSL values for Swt" do
-        path = double('path')
-        allow(::Swt::Path).to receive(:new) { path }
-        args = [100, 200, width, height, 180.0, -90.0]
-        expect(path).to receive(:add_arc).with(*args)
-        sw = 10
-        args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, 180, -90.0]
-        expect(gc).to receive(:draw_arc).with(*args)
-        subject.paint_control(gc)
-      end
+      sw = 10
+      args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, -180, -270.0]
+      expect(gc).to receive(:fill_path).with(path)
+      expect(gc).to receive(:draw_arc).with(*args)
+      subject.paint_control(gc)
+    end
+  end
+
+  context "wedge fill style" do
+    before :each do
+      allow(shape).to receive_messages(wedge?: true)
     end
 
-    context "wedge fill style" do
-      before :each do
-        allow(shape).to receive_messages(wedge?: true)
-      end
-
-      specify "fills arc" do
-        expect(gc).to receive(:fill_arc)
-        subject.paint_control(event)
-      end
-
-      specify "draws arc" do
-        expect(gc).to receive(:draw_arc)
-        subject.paint_control(event)
-      end
-
-      specify "translates DSL values for Swt" do
-        args = [100, 200, width, height, 180, -90.0]
-        expect(gc).to receive(:fill_arc).with(*args)
-        sw = 10
-        args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, 180, -90.0]
-        expect(gc).to receive(:draw_arc).with(*args)
-        subject.paint_control(gc)
-      end
+    specify "fills arc" do
+      expect(gc).to receive(:fill_arc)
+      subject.paint_control(event)
     end
+
+    specify "draws arc" do
+      expect(gc).to receive(:draw_arc)
+      subject.paint_control(event)
+    end
+
+    specify "translates DSL values for Swt" do
+      args = [left, top, width, height, -180, -270.0]
+      expect(gc).to receive(:fill_arc).with(*args)
+
+      sw = 10
+      args = [100 + sw / 2, 200 + sw / 2, width - sw, height - sw, -180, -270.0]
+      expect(gc).to receive(:draw_arc).with(*args)
+      subject.paint_control(gc)
+    end
+  end
+
+  describe "half-way around" do
+    #
+    #   ---------
+    #  /         \
+    # |           |
+    # |...........|
+    # |...........|
+    #  \........./
+    #   ---------
+    #
```

Comment:
So they're trying to visualize the dotted region as the part of the arc that we'll sweep through. This ones starts over on the right midpoint, then runs over to the left midpoint.

Any thoughts on how to convey this that would be clearer? I've found these type of drawings to be critical to me understanding the behavior as I work through it, but I admit I'm a bit deep in it at the moment so not surprised my scrawlings aren't totally clear :)